Changed folder structure and made packages for specific use case.

srrathi opened the pull request:
- [ ] To add tests
- [x] Proper commenting before every function.
- [ ] To add a go linter

srrathi committed:
updated with steps to test locally

srrathi committed:
added listing of channels in sidebar

added intial styling of the TUI and listing of channels in sidebar

srrathi committed:
added utility function for UI and logs printing

Added function to get First letter of string for showing Name First Letter in coloured box in UI and added function to print logs in a seperate file while TUI is rendering in terminal

srrathi committed:
added message sending, switching between channels and load previous message from history

srrathi committed:
added realtime loading of user message in chat and User Login Screen

srrathi committed:
added caching of user token and session login using it

srrathi committed:
added realtime updation of messages in channel

srrathi committed:
added token expiration time in cache

srrathi committed:
added loading of past messages

srrathi committed:
some code cleanup

srrathi committed:
added listing, filtering and sending of slash command

srrathi committed:
some code clean up

srrathi committed:
added password hiding on typing

srrathi committed:
added @ mentioning of channel members while sending message

srrathi committed:
some helper functions for adding highlighting of username in msg string

srrathi committed:
code cleanup - added handle for login screen update

srrathi committed:
added unified key bindings and improved its code structuring

srrathi committed:
resolved channel messages not listing on relogin after logout

srrathi committed:
seperate package for all the key bindings

srrathi committed:
made sepearate package for styles, cache and ui

srrathi committed:
moved all TUI code to respective packages and added flag to print logs in logger file

srrathi committed:
imporved help message

srrathi committed:
removed old files

srrathi committed:
exclude debug file and local sdk from commiting

srrathi committed:
Updated readme with TUI info

debdutdeb commented on ui/utils.go:
```diff
@@ -0,0 +1,73 @@
+	"github.com/joho/godotenv"
+)
+
+func goDotEnvVariable(key string) string {
+
+	// load .env file
+	err := godotenv.Load(".env")
+
+	if err != nil {
+		log.Fatalf("Error loading .env file")
+	}
+
+	return os.Getenv(key)
+}
```
Load the file once in `init` - then maybe get each key just once and export. Like,
```go
func init() {
    if err := godotenv.Load(".env"); err != nil {
        // handle error
    }
}
var SITE_URL = os.Getenv("DEV_SERVER_URL")
var EMAIL = os.Getenv("EMAIL")
var PASS = os.Getenv("PASS")
```
now just reuse those.

srrathi replied:
yes it shouldn't be loaded everytime I call the function. I'll update it.

debdutdeb commented on ui/utils.go:
```diff
@@ -0,0 +1,73 @@
+	return os.Getenv(key)
+}
+
+func getServerUrl() string {
+	DEV_SERVER_URL := goDotEnvVariable("DEV_SERVER_URL")
+	return DEV_SERVER_URL
+}
```
As said above you don't need this.

debdutdeb reviewed (changes requested):
Just a quick review on a sunday. As soon as I actually run it locally, I'll leave another one.

srrathi committed:
removed function which load .env everytime it is called

added flag to select work environment among development and production and server URL will be set according to it

srrathi committed:
updated readme with prod flag usage details

srrathi committed:
added hiding of list of suggested usernames if entered username is not available

srrathi committed:
if token is not valid show login screen to login

srrathi committed:
added adding of server url as a flag value at the time of running TUI

srrathi committed:
added commenting and optimised functions

srrathi committed:
refactored code and added commenting

srrathi committed:
updated readme to add server url as a flag while running main file

srrathi commented:
@debdutdeb I think we can merge it for now. If found any bugs and error we can open issue for it.

debdutdeb commented:
Sounds good @srrathi :)

debdutdeb commented:
Ah I can't merge this, @geekgonecrazy ping

srrathi committed:
added error handling for login screen

srrathi commented:
> Ah I can't merge this, @geekgonecrazy ping

@debdutdeb I can merge it. But may be @geekgonecrazy could give a quick feedback, we can merge after it. Wys
